src/lsp/request/generalRequest.ts

```diff
@@ -0,0 +1,75 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as vscode from "vscode";
+import { ChildProcess } from "child_process";
+import { IJSONRPCResponse, logToChannel } from "../response/generalResponse";
+
+export type IJSONRPCMethod =
+  | "initialize"
+  | "shutdown"
+  | "exit"
+  | "suggest/environmentVariables";
+
+export interface ILSPRequestDetails {
+  process: ChildProcess;
+  id: number;
+  method: IJSONRPCMethod;
+  params?: any;
+}
+
+export interface IJSONRPCRequest {
+  jsonrpc: "2.0";
+  id: number;
+  method: IJSONRPCMethod;
+  params: any;
+}
+
+// Function to send requests to the Language Server Protocol and read responses
+export function sendRequestToLSPReadResponse(
```

Comment:
Simplify name - maybe something like: `askLSP` (and response type is clear by return type)

Reply:
okay